oauth2client/contrib/gce.py

```diff
@@ -183,12 +145,4 @@ def service_account_email(self):
             AttributeError, if the email can not be retrieved from the Google
             Compute Engine metadata service.
         """
-        if self._service_account_email is None:
-            failure, email = _get_service_account_email()
-            if failure is None:
-                self._service_account_email = email
-            else:
-                raise AttributeError('Failed to retrieve the email from the '
-                                     'Google Compute Engine metadata service',
-                                     failure, email)
-        return self._service_account_email
+        return self._metadata.get_service_account_info()['email']
```

Comment:
I mean the actual http request to the metadata server could fail (in theory), but service account json will always come back with an email field. I could enumerate the fields as a tuple in returning them from the metadata server to be more clear I suppose?